providers/azure.go

```diff
@@ -233,13 +236,44 @@ func getEmailFromJSON(json *simplejson.Json) (string, error) {
 	return email, err
 }
 
-// GetEmailAddress returns the Account email address
-func (p *AzureProvider) GetEmailAddress(ctx context.Context, s *sessions.SessionState) (string, error) {
+func getEmailFromIDToken(idToken string) (string, error) {
+	jwt := strings.Split(idToken, ".")
+	jwtData := strings.TrimSuffix(jwt[1], "=")
+	b, err := base64.RawURLEncoding.DecodeString(jwtData)
```

Comment:
See my note below on consolidating logic & common `*oidc.IDTokenVerifiers`.

BUT, if those aren't applicable, I'm not a fan of this logic. Manually tearing apart a JWT like this geels hacky.

Even if its a dummy `oidc.IDTokenVerifier` like in the unit tests, this would be much cleaner if we convert the tokens string to an actual `oidc.IDToken`. Here is a sample using an `oidc.IDTokenVerifier` where the `oidc.KeySet` we use just has a dummy implementation of the `VerifySignature` method:

https://github.com/oauth2-proxy/oauth2-proxy/blob/8bed7aafbdb2cb6241b9655e983dd54d54b50531/pkg/middleware/jwt_session_test.go#L81

Reply:
Or some better way to convert a `string` -> `oidc.IDToken` -- but I'm not sure there's a cleaner way, hence we have the mock KeySets in our tests to help create `oidc.IDTokens`

Reply:
i refactored to use `jwt` package to extract claims